lib/facter/framework/formatters/legacy_fact_formatter.rb

```diff
@@ -84,6 +86,11 @@ def remove_enclosing_accolades(pretty_fact_json)
       pretty_fact_json.gsub(/^},/, '}')
     end
 
+    def remove_newlines(pretty_fact_json)
```

Comment:
i think this name is not very accurate, you are not removing newlines, you are replacing them with actual newlines. Please rename this to something more descriptive.